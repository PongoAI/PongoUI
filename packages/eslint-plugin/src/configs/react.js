/** @type {import("eslint").Linter.Config} */
const config = {
  extends: ['airbnb-typescript', 'prettier'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: 'tsconfig.json',
  },
  plugins: ['deprecation', 'import', '@typescript-eslint', 'jest', 'jsx-a11y', 'react', 'react-hooks'],
  settings: {
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      typescript: {
        alwaysTryTypes: true,
      },
    },
    'no-unused-expressions': 'off',
  },
  env: {
    browser: true,
    es6: true,
    node: true,
  },

  ignorePatterns: ['coverage', 'dist', 'dist-storybook', 'lib', 'lib-commonjs', '**/__snapshots__', 'node_modules'],
  rules: {
    'no-unused-vars': 'error',
    'no-use-before-define': 'off',
    '@typescript-eslint/no-use-before-define': ['error'],
    'import/no-extraneous-dependencies': ['error', { devDependencies: true }],
    'import/extensions': [
      'error',
      'ignorePackages',
      {
        js: 'never',
        jsx: 'never',
        ts: 'never',
        tsx: 'never',
      },
    ],
  },
};

module.exports = config;
