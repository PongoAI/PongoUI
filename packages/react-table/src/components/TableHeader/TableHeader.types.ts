import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';

export type TableHeaderSlots = {
  /**
   * The root of the TableHeader.
   */
  root: Slot<'thead'>;
};

type TableHeaderCommons = {};

export type TableHeaderProps = ComponentProps<TableHeaderSlots> & TableHeaderCommons;

export type TableHeaderState = ComponentState<TableHeaderSlots> & TableHeaderCommons;
