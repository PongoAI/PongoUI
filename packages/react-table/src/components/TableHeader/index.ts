export * from './TableHeader';
export * from './useTableHeaderStyles';
export * from './TableHeader.types';
export * from './useTableHeader';
