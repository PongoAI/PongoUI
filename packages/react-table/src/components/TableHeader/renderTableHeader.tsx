import * as React from 'react';
import { getSlots } from '@fluentui/react-utilities';
import type { TableHeaderState, TableHeaderSlots } from './TableHeader.types';

export const renderTableHeader = (state: TableHeaderState) => {
  const { slots, slotProps } = getSlots<TableHeaderSlots>(state);

  return <slots.root {...slotProps.root} />;
};
