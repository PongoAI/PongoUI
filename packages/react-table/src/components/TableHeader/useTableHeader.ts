import * as React from 'react';
import { getNativeElementProps } from '@fluentui/react-utilities';
import type { TableHeaderProps, TableHeaderState } from './TableHeader.types';

export const useTableHeader = (props: TableHeaderProps, ref: React.Ref<HTMLElement>): TableHeaderState => {
  const state: TableHeaderState = {
    components: {
      root: 'thead',
    },
    root: getNativeElementProps('thead', {
      ref,
      ...props,
    }),
  };

  return state;
};
