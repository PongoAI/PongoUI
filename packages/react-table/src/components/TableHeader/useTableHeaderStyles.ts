import { makeStyles, mergeClasses } from '@griffel/react';
import type { TableHeaderState } from './TableHeader.types';
import { tokens } from '@cebus/react-theme';

const useRootStyles = makeStyles({
  root: {
    display: 'table-header-group',
    verticalAlign: 'middle',
    fontWeight: tokens.bold,
  },
});

export const useTableHeaderStyles = (state: TableHeaderState) => {
  const rootStyles = useRootStyles();

  state.root.className = mergeClasses(rootStyles.root, state.root.className);

  return state;
};
