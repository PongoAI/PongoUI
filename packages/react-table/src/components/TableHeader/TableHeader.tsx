import * as React from 'react';
import { useTableHeader } from './useTableHeader';
import { useTableHeaderStyles } from './useTableHeaderStyles';
import { renderTableHeader } from './renderTableHeader';
import type { TableHeaderProps } from './TableHeader.types';
import type { ForwardRefComponent } from '@fluentui/react-utilities';

/**
 * The TableHeader component is used to display the header content for the Table.
 */
export const TableHeader: ForwardRefComponent<TableHeaderProps> = React.forwardRef((props, ref) => {
  const state = useTableHeader(props, ref);

  useTableHeaderStyles(state);

  return renderTableHeader(state);
});
TableHeader.displayName = 'TableHeader';
