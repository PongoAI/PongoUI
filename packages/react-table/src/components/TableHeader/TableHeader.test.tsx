import * as React from 'react';
import { TableHeader } from './TableHeader';
import { render } from '@testing-library/react';
import { isConformant } from '../../common/isConformant';

describe('TableHeader', () => {
  isConformant({
    Component: TableHeader,
    displayName: 'TableHeader',
  });

  it('renders a basic Table', () => {
    const renderedComponent = render(<TableHeader />);

    expect(renderedComponent.baseElement).toMatchSnapshot();
  });
});
