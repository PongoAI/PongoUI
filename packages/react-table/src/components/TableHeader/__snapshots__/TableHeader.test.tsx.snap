// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader renders a basic Table 1`] = `
<body>
  <div>
    <thead
      class="___p8zssc0 f1mldhui fmrv4ls f1wug5y4"
    />
  </div>
</body>
`;
